diff --git a/repos/extra-x86_64/PKGBUILD b/repos/extra-x86_64/PKGBUILD
index 40cd964..4f3dea2 100644
--- a/repos/extra-x86_64/PKGBUILD
+++ b/repos/extra-x86_64/PKGBUILD
@@ -32,3 +32,12 @@ build() {
 package() {
   DESTDIR="$pkgdir" cmake --install build
 }
+
+source+=('highlight-window-no-partial-opacity.patch')
+
+sha256sums+=('2e039adb85beca1717aa708ca7cc3febda66469113aba09f1b870518390b3b53')
+
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i ../highlight-window-no-partial-opacity.patch
+}
