diff --git a/repos/extra-x86_64/PKGBUILD b/repos/extra-x86_64/PKGBUILD
index e559392..50a6e81 100644
--- a/repos/extra-x86_64/PKGBUILD
+++ b/repos/extra-x86_64/PKGBUILD
@@ -13,15 +13,22 @@ depends=(kscreenlocker xcb-util-cursor plasma-framework kcmutils kwayland-server
 makedepends=(extra-cmake-modules qt5-tools kdoctools)
 optdepends=('qt5-virtualkeyboard: virtual keyboard support for kwin-wayland')
 groups=(plasma)
-source=("https://download.kde.org/stable/plasma/$pkgver/$pkgname-$pkgver.tar.xz"{,.sig})
+source=("https://download.kde.org/stable/plasma/$pkgver/$pkgname-$pkgver.tar.xz"{,.sig}
+        "highlight-window-no-partial-opacity.patch")
 install=$pkgname.install
 sha256sums=('36afa37b773d31c0bcb8f4e32fa1f06cc33e15ad00e729ba36f120bbe034903b'
-            'SKIP')
+            'SKIP'
+            '2e039adb85beca1717aa708ca7cc3febda66469113aba09f1b870518390b3b53')
 validpgpkeys=('2D1D5B0588357787DE9EE225EC94D18F7F05997E'  # Jonathan Riddell <jr@jriddell.org>
               '0AAC775BB6437A8D9AF7A3ACFE0784117FBCE11D'  # Bhushan Shah <bshah@kde.org>
               'D07BD8662C56CB291B316EB2F5675605C74E02CF'  # David Edmundson <davidedmundson@kde.org>
               '1FA881591C26B276D7A5518EEAAF29B42A678C20') # Marco Martin <notmart@gmail.com>
 
+prepare() {
+  cd "$pkgname-$pkgver"
+  patch -Np1 -i ../highlight-window-no-partial-opacity.patch
+}
+
 build() {
   cmake -B build -S $pkgname-$pkgver \
     -DCMAKE_INSTALL_LIBEXECDIR=lib \
